### PR TITLE
claiming: add claim own profile option

### PR DESCRIPTION
**Pull request by MJedr**
ref: #2402

### CODE REVIEW

File: ui/src/actions/actionTypes.js [hunk 1]
```diff
@@ -68,6 +68,10 @@ export const AUTHOR_PUBLICATION_SELECTION_CLEAR =
   'AUTHOR_PUBLICATION_SELECTION_CLEAR';
 export const AUTHOR_SET_ASSIGN_DRAWER_VISIBILITY =
   'AUTHOR_SET_ASSIGN_DRAWER_VISIBILITY';
+export const AUTHOR_PUBLICATION_CLAIMED_SELECTION =
```
Comment (drjova) [inline ui/src/actions/actionTypes.js:71]:
```suggestion
export const AUTHOR_PUBLICATION_CLAIM_SELECTION =
```

i would say this fits better as naming (applies to all :))

File: ui/src/authors/assignNotification.jsx [hunk 2]
```diff
@@ -39,6 +39,17 @@ export function assignSuccess({ from, to, papers }) {
   });
 }
 
+export function assignSuccessOwnProfile({ isUnassignAction }) {
+  const message = isUnassignAction
+    ? 'All selected papers will be removed from your profile.'
+    : 'All selected papers will be claimed to your profile..';
```
Comment (drjova) [inline ui/src/authors/assignNotification.jsx:45]:
```suggestion
    : 'All selected papers will be claimed to your profile.';
```

File: ui/src/reducers/authors.js [hunk 3]
```diff
@@ -19,6 +21,7 @@ import {
 export const initialState = Map({
   publicationSelection: Set(),
   isAssignDrawerVisible: false,
+  publicationSelectionClaimed: Map({}),
```
Comment (drjova) [inline ui/src/reducers/authors.js:24]: Can this be a `Set`? I don't see why it's different from the `publicationSelection`

Comment (MJedr) [reply ui/src/reducers/authors.js:24]: cause there is also `claimed` value, and each claimed has to be matched with corresponding paperId

File: ui/src/actions/authors.js [hunk 4]
```diff
@@ -5,13 +5,16 @@ import {
   AUTHOR_PUBLICATION_SELECTION_CLEAR,
   AUTHOR_PUBLICATION_SELECTION_SET,
   AUTHOR_SET_ASSIGN_DRAWER_VISIBILITY,
+  AUTHOR_PUBLICATION_CLAIM_SELECTION,
+  AUTHOR_PUBLICATIONS_CLAIMED_CLEAR,
```
Comment (drjova) [inline ui/src/actions/authors.js:9]:
```suggestion
  AUTHOR_PUBLICATIONS_CLAIM_CLEAR,
```

File: ui/src/actions/authors.js [hunk 5]
```diff
@@ -38,6 +41,19 @@ export function clearPublicationSelection() {
...
+export function setPublicationsClaimedSelection(papersIds, claimed, selected) {
+  return {
+    type: AUTHOR_PUBLICATION_CLAIM_SELECTION,
+    payload: { papersIds, claimed, selected },
+  };
+}
+
+export function clearPublicationsClaimedSelection() {
+  return {
+    type: AUTHOR_PUBLICATIONS_CLAIMED_CLEAR,
```
Comment (drjova) [inline ui/src/actions/authors.js:53]:
```suggestion
    type: AUTHOR_PUBLICATIONS_CLAIM_CLEAR,
```

**Review by drjova — APPROVED** [review]
💯 please don't forget to deactivate the feature flag in prod kubernetes :)

**Comment by MJedr** [comment]
@drjova for prod only or for prod and qa?

**Comment by drjova** [comment]
we have to disable it only on prod, I think QA should be available for testing.

**Pull request merged**